token-lending/program/src/state/last_update.rs

```diff
@@ -0,0 +1,58 @@
+use crate::error::LendingError;
+use solana_program::{clock::Slot, program_error::ProgramError};
+use std::cmp::Ordering;
+
+/// Number of slots to consider stale after
+pub const STALE_AFTER_SLOTS: u64 = 0;
+
+/// Last update state
+#[derive(Clone, Debug, Default)]
+pub struct LastUpdate {
+    /// Last slot when updated
+    pub slot: Slot,
+    /// True when marked stale, false when slot updated
+    pub stale: bool,
+}
+
+impl LastUpdate {
+    /// Create new last update
+    pub fn new(slot: Slot) -> Self {
+        Self { slot, stale: true }
+    }
+
+    /// Return slots elapsed since given slot
+    pub fn slots_elapsed(&self, slot: Slot) -> Result<u64, ProgramError> {
+        let slots_elapsed = slot
+            .checked_sub(self.slot)
+            .ok_or(LendingError::MathOverflow)?;
+        Ok(slots_elapsed)
+    }
+
+    /// Set last update slot
+    pub fn update_slot(&mut self, slot: Slot) {
+        self.slot = slot;
+        self.stale = false;
+    }
+
+    /// Set stale to true
+    pub fn mark_stale(&mut self) {
+        self.stale = true;
+    }
+
+    /// Check if last update slot is too long ago
+    pub fn is_stale(&self, slot: Slot) -> Result<bool, ProgramError> {
+        Ok(self.stale || self.slots_elapsed(slot)? > STALE_AFTER_SLOTS)
```

Comment:
`STALE_AFTER_SLOTS` is currently set to `0`, so a reserve or obligation is considered stale outside of the current slot or if marked stale. If it's the current slot and we use `>= 0` it would always be stale.

Reply:
If it's currently set to 0, that reads to me like the value is "stale after 0 slots" which is less clear to me than something like "stale after 1 slot" or "fresh for 1 slot"

Reply:
I guess the current behavior is that it's only fresh for the current slot, so it would be "stale _at_ 1 slot". I'll think about this more though and revisit after testing to even see if executing everything in the current slot is reasonable.

Reply:
To be clear, I agree with the current behavior. I was just commenting on code readability. Setting `STALE_AFTER_SLOTS` to `1` and switching this to `>= STALE_AFTER_SLOTS` would make me happy 😜

Reply:
Done.